flixel/tile/FlxBaseTilemap.hx

```diff
@@ -609,38 +610,79 @@ class FlxBaseTilemap<Tile:FlxObject> extends FlxObject
 	}
 
 	/**
-	 * Adjust collision settings and/or bind a callback function to a range of tiles.
-	 * This callback function, if present, is triggered by calls to overlap() or overlapsWithCallback().
+	 * Add a new Tile Filter to this tilemap which allows you to setup specific collision and/or callback based on class.
+	 * Use `removeTileFilter` to remove the filter.
 	 * 
-	 * @param	Tile				The tile or tiles you want to adjust.
-	 * @param	AllowCollisions		Modify the tile or tiles to only allow collisions from certain directions, use FlxObject constants NONE, ANY, LEFT, RIGHT, etc. Default is "ANY".
-	 * @param	Callback			The function to trigger, e.g. lavaCallback(Tile:FlxObject, Object:FlxObject).
-	 * @param	CallbackFilter		If you only want the callback to go off for certain classes or objects based on a certain class, set that class here.
-	 * @param	Range				If you want this callback to work for a bunch of different tiles, input the range here. Default value is 1.
+	 * @param	Tile		The tile id that this filter is for, or the first tile id in a range of tiles.
+	 * @param	Filter		Specify a class for this filter to work with, defaults to `FlxObject`
+	 * @param	Collisions	Specify Collision flags to be used for this tile. Defaults to ANY
+	 * @param	Callback	The function to trigger when overlap is detected. e.g. `tileCallback(Tile:FlxObject, Object:FlxObject)`
+	 * @param	ProcessCallback	This function, if set, will be called when overlap is detected, and will return true
+	 * @param	Range		If you want this filter to be applied to a range of different tiles, input the range here. Default is 1.
 	 */
-	public function setTileProperties(Tile:Int, AllowCollisions:Int = FlxObject.ANY, ?Callback:FlxObject->FlxObject->Void, ?CallbackFilter:Class<FlxObject>, Range:Int = 1):Void
+	public function addTileFilter(Tile:Int, ?Filter:Class<FlxObject>, ?Collisions:Int = FlxObject.ANY, ?Callback:FlxTile-> FlxObject->Void, ?ProcessCallback:FlxTile-> FlxObject->Bool, ?Range:Int = 1):Void
```

Comment:
Correct. I couldn't find a good way/reason to keep the old behavior in tact.
Not too worried that it can't be merged now - just want to make sure to keep it. I use this logic in several of my current projects and accidentally lost it.